models/paraswap/base/paraswap_v5_base_trades.sql

```diff
@@ -0,0 +1,117 @@
+{% set trade_event_tables = [
+    source('paraswap_base', 'AugustusSwapper_evt_BoughtV3')
+    ,source('paraswap_base', 'AugustusSwapper_evt_SwappedV3')
+] %}
+
+WITH dexs AS (
+    {% for trade_table in trade_event_tables %}
+        SELECT 
+            evt_block_time AS block_time,
+            evt_block_number AS block_number,
+            beneficiary AS taker, 
+            initiator AS maker, 
+            receivedAmount AS token_bought_amount_raw,
+            srcAmount AS token_sold_amount_raw,
+            CAST(NULL AS double) AS amount_usd,
+            CASE 
+                WHEN destToken = 0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
+                THEN 0x4200000000000000000000000000000000000006 -- WETH 
+                ELSE destToken
+            END AS token_bought_address,
+            CASE 
+                WHEN srcToken = 0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
+                THEN 0x4200000000000000000000000000000000000006 -- WETH 
+                ELSE srcToken
+            END AS token_sold_address,
+            contract_address AS project_contract_address,
+            evt_tx_hash AS tx_hash, 
+            CAST(ARRAY[-1] as array<bigint>) AS trace_address,
+            evt_index
+        FROM {{ trade_table }} p 
+        {% if is_incremental() %}
+        WHERE p.evt_block_time >= date_trunc('day', now() - interval '7' day)
+        {% endif %}
+        {% if not loop.last %}
+        UNION ALL
+        {% endif %}
+    {% endfor %}
+)
+
+SELECT 'base' AS blockchain,
+    'paraswap' AS project,
+    '5' AS version,
+    cast(date_trunc('day', d.block_time) as date) as block_date,
+    cast(date_trunc('month', d.block_time) as date) as block_month,
+    d.block_time,
+    e1.symbol AS token_bought_symbol,
+    e2.symbol AS token_sold_symbol,
+    CASE
+        WHEN lower(e1.symbol) > lower(e2.symbol) THEN concat(e2.symbol, '-', e1.symbol)
+        ELSE concat(e1.symbol, '-', e2.symbol)
+    END AS token_pair,
+    d.token_bought_amount_raw / power(10, e1.decimals) AS token_bought_amount,
+    d.token_sold_amount_raw / power(10, e2.decimals) AS token_sold_amount,
+    d.token_bought_amount_raw,
+    d.token_sold_amount_raw,
+    coalesce(
+        d.amount_usd
+        ,(d.token_bought_amount_raw / power(10, p1.decimals)) * p1.price
+        ,(d.token_sold_amount_raw / power(10, p2.decimals)) * p2.price
+    ) AS amount_usd,
+    d.token_bought_address,
+    d.token_sold_address,
+    coalesce(d.taker, tx."from") AS taker,
+    d.maker,
+    d.project_contract_address,
+    d.tx_hash,
+    tx."from" AS tx_from,
+    tx.to AS tx_to,
+    d.trace_address,
+    d.evt_index
+FROM dexs d
+INNER JOIN {{ source('base', 'transactions') }} tx ON d.tx_hash = tx.hash
+    AND d.block_number = tx.block_number
+    {% if not is_incremental() %}
+    AND tx.block_time >= TIMESTAMP '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+    AND tx.block_time >= date_trunc('day', now() - interval '7' day)
```

Comment:
We may use `incremental_predicate` for the incremental filters:
```suggestion
    AND {{incremental_predicate('tx.block_time')}}
```

Reply:
thank you @Hosuke, I'll fix this now.